### PR TITLE
V5 Topologies api should return timestamps in rfc3339 format

### PR DESCRIPTION
**Pull request by srijeet0406**
This PR is not related to any issue. It fixes the `topologies` api in version 5.0, so that it returns rfc3339 formatted timestamps. 

<hr/>

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Ops

## What is the best way to verify this PR?

Make sure all the tests pass.
Make sure the CRUD routes on topologies now return rfc3339 timestamps.
## PR submission checklist
- [x] This PR has tests 
- [x] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Comment by srijeet0406** [comment]
Looking into the failing tests.

File: traffic_ops/traffic_ops_golang/topology/validation.go [hunk 1]
```diff
@@ -54,6 +56,45 @@ func checkForSelfParents(nodes []tc.TopologyNode, index int) error {
 	return nil
 }
 
+// checkForEdgeParents returns an error if an index given in the parents array, adds a warning + returns a nil error if
```
Comment (shamrickus) [inline traffic_ops/traffic_ops_golang/topology/validation.go:59]: The first part of this comment `returns an error if an index given in the parents array, ...` doesn't make sense.

File: traffic_ops/traffic_ops_golang/topology/topologies.go [hunk 2]
```diff
@@ -419,6 +591,286 @@ func (topology *TOTopology) Create() (error, error, int) {
 	return nil, nil, 0
 }
 
+func readTopologies(r *http.Request, useIMS bool) ([]interface{}, error, error, int, *time.Time) {
+	var maxTime time.Time
+	var runSecond bool
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		//api.HandleErr(w, r, tx, errCode, userErr, sysErr)
```
Comment (shamrickus) [inline traffic_ops/traffic_ops_golang/topology/topologies.go:599]: Commented line should be removed.

File: traffic_ops/traffic_ops_golang/topology/topologies.go [hunk 3]
```diff
@@ -419,6 +591,286 @@ func (topology *TOTopology) Create() (error, error, int) {
...
+		var parents pq.Int64Array
+		if err = rows.Scan(
+			&name,
+			&description,
+			&lastUpdated,
+			&topologyNode.Id,
+			&topologyNode.Cachegroup,
+			&parents,
+		); err != nil {
+			//api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, errors.New("topology read: scanning: "+err.Error()))
```
Comment (shamrickus) [inline traffic_ops/traffic_ops_golang/topology/topologies.go:655]: Commented line should be removed.

File: traffic_ops/traffic_ops_golang/topology/topologies.go [hunk 4]
```diff
@@ -419,6 +591,286 @@ func (topology *TOTopology) Create() (error, error, int) {
...
+		}
+		indices[topologyNode.Id] = index
+		if _, exists := topologies[name]; !exists {
+			topology := tc.TopologyV5{Nodes: []tc.TopologyNodeV5{}}
+			topologies[name] = &topology
+			topology.Name = name
+			topology.Description = description
+			topology.LastUpdated, err = util.ConvertTimeFormat(lastUpdated, time.RFC3339)
+			if err != nil {
+				//api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, errors.New("couldn't convert last updated time to rfc3339 format: "+err.Error()))
```
Comment (shamrickus) [inline traffic_ops/traffic_ops_golang/topology/topologies.go:669]: Commented line should be removed.

File: traffic_ops/traffic_ops_golang/topology/topologies.go [hunk 5]
```diff
@@ -419,6 +591,286 @@ func (topology *TOTopology) Create() (error, error, int) {
...
+			return
+		}
+	}
+
+	if userErr, sysErr, statusCode := setTopologyDetails(tx, &topology); userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, statusCode, userErr, sysErr)
+		return
+	}
+
+	if userErr, sysErr, statusCode := addNodes(tx, topology.Name, topology.Nodes); userErr != nil || sysErr != nil {
```
Comment (shamrickus) [inline traffic_ops/traffic_ops_golang/topology/topologies.go:855]: These vars don't need to be redeclared.

File: traffic_ops/traffic_ops_golang/topology/topologies.go [hunk 6]
```diff
@@ -419,6 +591,286 @@ func (topology *TOTopology) Create() (error, error, int) {
...
+		api.HandleErr(w, r, tx, statusCode, userErr, sysErr)
+		return
+	}
+	if userErr, sysErr, statusCode := addParents(tx, topology.Nodes); userErr != nil || sysErr != nil {
```
Comment (shamrickus) [inline traffic_ops/traffic_ops_golang/topology/topologies.go:859]: These vars don't need to be redeclared.

File: traffic_ops/testing/api/v5/topologies_test.go [hunk 7]
```diff
@@ -470,42 +469,42 @@ func TestTopologies(t *testing.T) {
...
-						t.Run(name, func(t *testing.T) {
-							if _, ok := testCase.RequestOpts.QueryParameters["name"]; !ok {
-								t.Fatalf("Query Parameter: \"name\" is required for DELETE method tests.")
-							}
-							alerts, reqInf, err := testCase.ClientSession.DeleteTopology(testCase.RequestOpts.QueryParameters["name"][0], testCase.RequestOpts)
-							for _, check := range testCase.Expectations {
-								check(t, reqInf, nil, alerts, err)
-							}
-						})
+					//case "PUT":
```
Comment (shamrickus) [inline traffic_ops/testing/api/v5/topologies_test.go:486]: This test should either not be commented out or the the commented out line should be removed.

File: traffic_ops/traffic_ops_golang/topology/topologies.go [hunk 8]
```diff
@@ -21,8 +21,10 @@ package topology
 
 import (
 	"database/sql"
+	"encoding/json"
 	"errors"
 	"fmt"
+	"github.com/apache/trafficcontrol/lib/go-rfc"
```
Comment (shamrickus) [inline traffic_ops/traffic_ops_golang/topology/topologies.go:27]: Local imports should go below system imports

**Review by shamrickus — CHANGES_REQUESTED** [review]
Functionality checks out, just some small comments.

**Review by shamrickus — APPROVED** [review]
LGTM

**Pull request merged**